worker/src/common/cryptography.py

```diff
@@ -0,0 +1,67 @@
+import base64
+import datetime
+from dataclasses import dataclass
+from pathlib import Path
+
+from cryptography.exceptions import UnsupportedAlgorithm
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey
+from cryptography.hazmat.primitives.asymmetric.types import (
+    PrivateKeyTypes,
+)
+from cryptography.hazmat.primitives.serialization import SSHPrivateKeyTypes
+
+
+@dataclass
+class AuthMessage:
+    """An authentication message for a worker"""
+
+    body: str
+    signature: str
+
+
+def load_private_key_from_path(private_key_fpath: Path) -> RSAPrivateKey:
```

Comment:
Is it too much work to also support EDCSA? I thought it was ready on backend side, but I might have misunderstood you. Not a big game changer, we can postpone this to later.

Reply:
It's not too much work. I can do it in a new PR.